### PR TITLE
Aod update dev

### PR DESCRIPTION
**Pull request by ircwaves**
MODIS AOD Collection 6 

Note that long term average values are updated outside of gips, and should be brought into gips for consistency.

### CODE REVIEW

File: gips/data/aod/aod.py [hunk 1]
```diff
@@ -47,7 +51,7 @@ def data_path(cls, tile='', date=''):
 
     @classmethod
     def find_tiles(cls):
-        return ['']
+        return ['h01v01']
```
Comment (ags-tolson) [inline gips/data/aod/aod.py]: Optional:  I found it handy in the sentinel-2 driver to set a module attribute `asset_type = 'L1C'`.  Then the compiler catches spelling errors for me and the error messages are obvious.  You could do that here if you want, call it `singleton_tid` or similar.

File: gips/data/aod/aod.py [hunk 2]
```diff
@@ -92,14 +97,22 @@ def __init__(self, filename):
...
         self.asset = bname[0:5]
-        self.tile = ''
+        self.tile = 'h01v01'
         year = bname[10:14]
         doy = bname[14:17]
         self.date = datetime.datetime.strptime(year + doy, "%Y%j").date()
         self.sensor = bname[:3]
-        #datafiles = self.datafiles()
+        cnum = float(bname[18:21])
+        self.modis_collection = cnum / 10 ** numpy.floor(numpy.log10(cnum))
```
Comment (ags-tolson) [inline gips/data/aod/aod.py]: optional:  This line may warrant a comment just to explain what's going on.

File: gips/data/aod/aod.py [hunk 3]
```diff
@@ -151,24 +164,61 @@ class aodData(Data):
...
+
 
     @classmethod
     def process_composites(cls, inventory, products, **kwargs):
+        '''
+        Currently inoperative do to absence of 'start_day' and 'end_day' from
+        DataInventory class.
+        '''
+        # since it is broken
+        raise Exception('Composite processing is currently broken')
```
Comment (ags-tolson) [inline gips/data/aod/aod.py]: optional:  Change to `NotImplementedError`

File: gips/data/aod/aod.py [hunk 4]
```diff
@@ -206,7 +256,12 @@ def process_mean(cls, filenames, fout):
             totalvar[inds] = numpy.divide(totalvar[inds], counts[inds])
             imgout[1].Write(totalvar)
             t = datetime.datetime.now() - start
-            VerboseOut('%s: mean/var for %s files processed in %s' % (os.path.basename(fout), len(filenames), t))
+            utils.verbose_out(
+                '%s: mean/var for %s files processed in %s' %
```
Comment (ags-tolson) [inline gips/data/aod/aod.py]: Optional:  Python 2 official docs says `"foo {} bar {}".format(...)` "should be preferred" over `"foo %s bar" % (...)"` in new code:  https://docs.python.org/2/library/stdtypes.html#str.format

File: gips/data/aod/aod.py [hunk 5]
```diff
@@ -206,7 +256,12 @@ def process_mean(cls, filenames, fout):
...
+                (os.path.basename(fout), len(filenames), t)
+            )
+        else:
+            raise Exception('No filenames provided')
```
Comment (ags-tolson) [inline gips/data/aod/aod.py]: optional:  Raise at the top of the method if there are no filenames, then the rest of the method can be moved out of the conditional.  However this keeps the diff smaller, which has value too.

**Comment by ags-tolson** [comment]
Okay all I could find are nits to pick, so :+1:

File: gips/data/aod/aod.py (hunk 1 above)
Comment (ircwaves) [inline gips/data/aod/aod.py]: noted.

File: gips/data/aod/aod.py (hunk 2 above)
Comment (ircwaves) [inline gips/data/aod/aod.py]: mathemagic

File: gips/data/aod/aod.py (hunk 3 above)
Comment (ircwaves) [inline gips/data/aod/aod.py]: agreed.

File: gips/data/aod/aod.py (hunk 4 above)
Comment (ircwaves) [inline gips/data/aod/aod.py]: Though I prefer the `str.format`, this one will slide.

**Comment by ircwaves** [comment]
So, in the sense that this updates the `aod` driver to have some cursory testing, and utilize MODIS collection 6, I think this is good to go.  @bhbraswell and I will discuss the `process_composites` update that is required for the driver to be fully functional.

**Pull request merged**